static/src/javascripts/projects/common/modules/check-dispatcher.js

```diff
@@ -0,0 +1,166 @@
+// @flow
+import config from 'lib/config';
+import { allEmailCanRun, listCanRun } from 'common/modules/email/run-checks';
+import emailArticle from 'common/modules/email/email-article';
+import clash from 'common/modules/experiments/ab-test-clash';
+import trackAdRender from 'commercial/modules/dfp/track-ad-render';
+import commercialFeatures from 'commercial/modules/commercial-features';
+import { checks } from './check-mediator-checks';
+import { resolveCheck, waitForCheck } from './check-mediator';
+
+const someCheckPassed = results => results.includes(true);
```

Comment:
I wasn't sure what the suggested practise was regarding types, I followed the approach that they're mandatory on public functions then optional on private functions? I'm happy to add type annotations if you'll feel they'll provide useful guidance for developers editing/reading this file!